public_src/components/sidebar/validation-browser.component.ts

```diff
@@ -1,30 +1,37 @@
-import { AppActions } from '../../store/app/actions';
-
-import { Component } from '@angular/core';
-import { select } from '@angular-redux/store';
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
 import { Observable } from 'rxjs';
 
+import { NgRedux, select } from '@angular-redux/store';
+import { IAppState } from '../../store/model';
+import { AppActions } from '../../store/app/actions';
+
 import { BsModalService } from 'ngx-bootstrap';
 
 import { ErrorHighlightService } from '../../services/error-highlight.service';
 import { MapService } from '../../services/map.service';
 import { OverpassService } from '../../services/overpass.service';
 import { StorageService } from '../../services/storage.service';
 import { ConfService } from '../../services/conf.service';
+
 import { IPtStop } from '../../core/ptStop.interface';
+import { ISuggestionsBrowserOptions } from '../../core/editingOptions.interface';
+import { IErrorObject, IRefErrorObject } from '../../core/errorObject.interface';
 
 @Component({
   selector: 'validation-browser',
   templateUrl: './validation-browser.component.html',
   styleUrls: ['./validation-browser.component.less'],
 })
-export class ValidationBrowserComponent {
+export class ValidationBrowserComponent implements OnInit, OnDestroy{
   @select(['app', 'editing']) public readonly editing$: Observable<boolean>;
-  @select(['app', 'errorCorrectionMode']) public readonly errorCorrectionMode$: Observable<string>;
+  @select(['app', 'errorCorrectionMode']) public readonly errorCorrectionMode$: Observable<object>;
   @select(['app', 'switchMode']) public readonly switchMode$: Observable<boolean>;
 
-  public refErrorsO;
-  public nameErrorsO;
+  public refErrorsO: IRefErrorObject[];
```

Comment:
I would prefer to see Obj at the end instead of one O letter. It was not obvious in the template...